### PR TITLE
Users/tihuang/diskusage

### PR DESCRIPTION
**Pull request by TingluoHuang**

### CODE REVIEW

**Comment by ericsciple** [comment]
nice

**Comment by bryanmacfarlane** [comment]
Can we print out at the beginning of a build if the drive where the work folder is on has 90%+ usage or < x GB free?  Chat with ChrisP for thoughts.

**Comment by bryanmacfarlane** [comment]
How expensive is it?  If real cheap, same check in between tasks might be interesting.

File: src/Agent.Worker/Build/TrackingManager.cs [hunk 1]
```diff
@@ -228,25 +228,57 @@ public void DisposeCollectedGarbage(IExecutionContext executionContext)
...
+                    {
+                        var gcConfig = LoadIfExists(executionContext, gcFile) as TrackingConfig;
+                        ArgUtil.NotNull(gcConfig, nameof(TrackingConfig));
 
-                    string fullPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), gcConfig.BuildDirectory);
-                    executionContext.Output(StringUtil.Loc("Deleting", fullPath));
-                    IOUtil.DeleteDirectory(fullPath, CancellationToken.None);
+                        string fullPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), gcConfig.BuildDirectory);
+                        executionContext.Output(StringUtil.Loc("Deleting", fullPath));
+                        IOUtil.DeleteDirectory(fullPath, executionContext.CancellationToken);
```
Comment (ericsciple) [inline src/Agent.Worker/Build/TrackingManager.cs]: Ideally we would also tf delete to cleanup AppData, C:\ProgramData, and server registration. We should add a story so it's at least a known limitation.

**Pull request merged**